lib/rspec/mocks/configuration.rb

```diff
@@ -3,12 +3,29 @@ module Mocks
     # Provides configuration options for rspec-mocks.
     class Configuration
       def initialize
+        @allow_message_expectations_on_nil = nil
         @yield_receiver_to_any_instance_implementation_blocks = true
         @verify_doubled_constant_names = false
         @transfer_nested_constants = false
         @verify_partial_doubles = false
       end
 
+      # Sets whether RSpec will warn, ignore, or fail a test when
+      # expectations are set on nil.
+      # By default, when this flag is not set, warning messages are issued when
+      # expectations are set on nil. This is to prevent false-positives and to
+      # catch potential bugs early on.
+      # When set to `true`, warning messages are suppressed.
+      # When set to `false`, it will raise an error.
+
```

Comment:
Thanks for taking the time to write up good docs for this!

This blank line causes YARD to not understand that this command block above is about `allow_message_expectations _on_nil`.  Here's how it renders right now:

![screen shot 2015-07-07 at 2 26 45 pm](https://cloud.githubusercontent.com/assets/49391/8557760/5035d634-24b4-11e5-8604-a64fbb7338f6.png)

I think if you add a line with only a `#` like this:

``` ruby
# Some comments
#
# Some more comments
```

...then YARD will render it properly.  You can check how YARD renders the docs by running `yard server --reload`.  It's generally a good idea to check that when writing the docs as their can be gotchas where things don't render how you expect.